src/app/features/location/gps.service.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { of } from "rxjs";
+import { GpsService } from "./gps.service";
+import { GeoService } from "./geo.service";
+import { MapPopupComponent } from "./map-popup/map-popup.component";
+
+function mockGeolocationPosition() {
+  return {
+    coords: {
+      latitude: 51.5074,
+      longitude: -0.1278,
+      accuracy: 10,
+      altitude: null,
+      altitudeAccuracy: null,
+      heading: null,
+      speed: null,
+    },
+    timestamp: Date.now(),
+  };
+}
+
+describe("GpsService", () => {
+  let service: GpsService;
+  let geoService: jasmine.SpyObj<GeoService>;
+  let mapPopupComponent: MapPopupComponent;
+  let fixture: ComponentFixture<MapPopupComponent>;
+  let location = { lat: 51.5074, lon: -0.1278 };
+
+  beforeEach(async () => {
+    const geoServiceSpy = jasmine.createSpyObj("GeoService", ["lookup"]);
+
+    TestBed.configureTestingModule({
+      declarations: [MapPopupComponent],
+      providers: [GpsService, { provide: GeoService, useValue: geoServiceSpy }],
+    })
```

Comment:
Any specific reason why you want to use the component in this test setup, @Ayush8923 ?

For a classic unit test, I would only create an instance of the service itself and test the service in isolation without any components.